lib/inspec/resources/groups.rb

```diff
@@ -200,7 +200,38 @@ def groups
   # implements generic unix groups via /etc/group
   class UnixGroup < GroupInfo
     def groups
-      inspec.etc_group.entries
+      get_group_info
+    end
+
+    private
+
+    def get_group_info
+      # check if getent is available on the system
+      bin = find_getent_utility
+
+      # fetch details of the groups using getent
+      cmd = inspec.command("#{bin} group")
+      raise Inspec::Exceptions::ResourceFailed, "Executing #{bin} group failed: #{cmd.stderr}" if cmd.exit_status.to_i != 0
+
+      # Split the output into lines and map each line to a hash
+      cmd.stdout.strip.split("\n").map do |line|
+        name, password, gid, members = line.split(":")
+        {
+          "name" => name,
+          "password" => password,
+          "gid" => gid.to_i, # Convert gid to an integer
+          "members" => members,
+        }
+      end
+    end
+
+    # check if getent exist in the system
+    def find_getent_utility
+      %w{/usr/bin/getent /bin/getent getent}.each do |cmd|
+        return cmd if inspec.command(cmd).exist?
+      end
+
+      raise Inspec::Exceptions::ResourceFailed, "Could not find `getent` on your system."
```

Comment:
If the `getent` utility is not available should we switch back to the old approach of using `inspec.etc_group.entries`?  And then log an error about the unavailability of `getent`?

Similarly, if there is an error encountered while using `getent` utility on line 213, should we again fallback to old approach and give it a try?

Reply:
Personally, I wouldn't mind having the fallback to fetch group information from `inspec.etc_group.entries`. Do you think we should implement this fallback?

Reply:
We could log the same exception message, appending an extension that indicates reading group information from `/etc/group` at the debug level, without raising an exception

Reply:
I feel that currently there might be some percentage of users already using this resource without `getent` utility. So it should not break for them if they upgrade.